### PR TITLE
[bug] create watched dir if it doesn't exist

### PR DESCRIPTION
**Pull request by kalikaneko**
- Resolves: #6936

### CODE REVIEW

**Comment by kalikaneko** [comment]
rebased this one.
awaiting for discussion on https://leap.se/code/issues/6936 to see if we should merge this or not.

**Comment by elijh** [comment]
two comments:

* what happens if the it is unable to create the directory?
* can you make the permissions not world readable/executable?

File: src/leap/mx/mail_receiver.py [hunk 1]
```diff
@@ -145,7 +146,7 @@ def _start_watching_dir(self, dirname, recursive):
         directory = filepath.FilePath(dirname)
         try:
             if not directory.isdir():
-                raise OSError("Not a directory: '%s'" % directory.path)
+                mkdir_p(dirname)
```
Comment (micah) [inline src/leap/mx/mail_receiver.py]: The permissions and ownership:group should be set here, owner:group (right now leap-mx is being run as root, but we don't want leap-mx to be running as root forever). Also, shouldn't this check to see that the directory creation actually happened?

**Comment by varac** [comment]
see https://leap.se/code/issues/6936#note-5 for permissions/ownership

@kali: did you tested this p.r. ? i doubt it will work in the current state.

**Comment by elijh** [comment]
also, leap_mx is running as root, so that directory gets created as owned by root. alas, we want leap_mx to not be run as root in the future. 

i don't think it makes sense to bother modifying the mkdir to make it change the owner. but, so long as leap_mx is run by root, then it is creating directories with a user that is not ideal.

**Comment by varac** [comment]
i mean with testing: 

- start leap_mx
- send a mail, should get delivered by postfix and by leap_mx

**Comment by kalikaneko** [comment]
> what happens if the it is unable to create the directory?
it will log the same error as before.
> can you make the permissions not world readable/executable?
sure.
what user should it use instead of root?
> did you tested
no. but since it's inside the try, any exception raised will be catched.

**Comment by elijh** [comment]
> what user should it use instead of root?

alas, not an easy answer. this is what would be ideal:

1. run leap_mx as a user that is configured (not hardcoded as vmail).
2. leap_mx does not bother to modify ownership of the Maildir

but, since it is running as root, i am not sure what is good. how about:

1. change the owner to the hardcoded value of 'vmail' if it is running as root.

**Comment by kalikaneko** [comment]
> change the owner to the hardcoded value of 'vmail' if it is running as root.

sounds good.

File: src/leap/mx/mail_receiver.py (hunk 1 above)
Comment (micah) [inline src/leap/mx/mail_receiver.py]: I have a pull request to change leap-mx from running as root: https://github.com/leapcode/leap_mx/pull/78

The leap platform will create the correct user/group before the package is installed.

**Comment by micah** [comment]
I think what we should do is:

1. Accept https://github.com/leapcode/leap_mx/pull/78 - this will run leap-mx as the user/group 'leap-mx' instead of root. It allows for these values to be changed by changing /etc/default/leap-mx
2. We create leap-mx users in the platform, it also creates the /var/mail/leap-mx directory, all before the package is installed
3. We do not accept this merge request, and instead leap-mx will continue to complain if the directory doesn't exist (this would be a legitimate error)
4. in the debian package, we create the user and directory, this is somewhat redundant, but it makes it so people can install leap-mx package without needing to deploy the platform.

**Pull request closed without merging**